src/segyio/segy.c

```diff
@@ -658,39 +658,36 @@ static int segy_sample_interval( segy_file* fp, double* dt) {
         return err;
     }
 
-    int _binary_header_dt;
-    segy_get_bfield(bin_header, BIN_Interval, &_binary_header_dt);
-    double binary_header_dt = _binary_header_dt/1000.0;
-
-    int _trace_header_dt;
-    segy_get_field(trace_header, TRACE_SAMPLE_INTERVAL, &_trace_header_dt);
-    double trace_header_dt = _trace_header_dt/1000.0;
-
-
-    *dt = trace_header_dt;
-
-    if (binary_header_dt == 0.0 && trace_header_dt == 0.0) {
-        fprintf(stderr, "Trace sampling rate in SEGY header and trace header set to 0.0. Will default to 4 ms.\n");
-        *dt = 4.0;
-    } else if (binary_header_dt == 0.0) {
-        fprintf(stderr, "Trace sampling rate in SEGY header set to 0.0. Will use trace sampling rate of: %f\n", trace_header_dt);
-        *dt = trace_header_dt;
-    } else if (trace_header_dt == 0.0) {
-        fprintf(stderr, "Trace sampling rate in trace header set to 0.0. Will use SEGY header sampling rate of: %f\n", binary_header_dt);
-        *dt = binary_header_dt;
-    } else if (trace_header_dt != binary_header_dt) {
-        fprintf(stderr, "Trace sampling rate in SEGY header and trace header are not equal. Will use SEGY header sampling rate of: %f\n",
-                binary_header_dt);
-        *dt = binary_header_dt;
+    // microseconds: us
+    int _binary_header_dt_us;
+    int _trace_header_dt_us;
+
+    segy_get_bfield(bin_header, BIN_Interval, &_binary_header_dt_us);
+    segy_get_field(trace_header, TRACE_SAMPLE_INTERVAL, &_trace_header_dt_us);
+
+    // milliseconds: ms
+    double binary_header_dt_ms = _binary_header_dt_us/1000.0;
+    double trace_header_dt_ms = _trace_header_dt_us/1000.0;
+
+    if (_binary_header_dt_us == 0 && _trace_header_dt_us == 0) {
+        fprintf(stderr, "Trace sampling rate in SEGY header and first trace header set to 0.0. Will not set.\n");
+    } else if (_binary_header_dt_us == 0) {
+        *dt = trace_header_dt_ms;
+    } else if (_trace_header_dt_us == 0) {
+        *dt = binary_header_dt_ms;
+    } else if (_binary_header_dt_us != _trace_header_dt_us) {
+        fprintf(stderr, "Trace sampling rate in SEGY header: %f and first trace header: %f are not equal. Will not set.\n",
+                binary_header_dt_ms, trace_header_dt_ms);
+    } else {
+        *dt = trace_header_dt_ms;
     }
 
     return 0;
 
 }
 
-int segy_sample_indexes( segy_file* fp, double* buf, double t0, size_t count) {
+int segy_sample_indexes( segy_file* fp, double* buf, double t0, double dt, size_t count) {
```

Comment:
agreed, but was somewhat mixed usage of double and float. t0 and sample_indexes is double. Can of course change all that.